tests/test_product_manager.py

```diff
@@ -0,0 +1,104 @@
+
+    @classmethod
+    def setUpClass(self):
+        self.database_connector = database_connector.DatabaseConnector()
+        self.item_manager = database_manager.ItemManager(
+            self.database_connector)
+        self.tag_manager = database_manager.TagManager(
+            self.database_connector)
+        aws_connector = AWSConnector()
+        self.s3_resource = aws_connector.get_s3_resource()
+        self.image_manager = ImageManager(self.s3_resource, self.bucket)
+        self.product_manager = product_manager.ProductManager(
+            self.item_manager, self.tag_manager, self.image_manager)
+
+    def setUp(self):
+        setup_test.create_test_database_setup(self.database_connector)
+        self.s3_resource.create_bucket(Bucket=self.bucket, CreateBucketConfiguration={
+            'LocationConstraint': 'eu-west-2'},)
+
+    def test_add_product(self):
+        # arrange
+        tags = ['bird', 'fineline']
+        vector_file = "tests/test2.jpg"
+        png_file = "tests/test2.jpg"
+
+        # act
+        self.product_manager.add_product(tags, vector_file, png_file)
+
+        # assert
+        self.assertTrue(setup_test.assertCount(
+            self.database_connector, 'Items', 1))
+        self.assertTrue(setup_test.assertCount(
+            self.database_connector, 'Tags', 2))
+
+    def test_delete_product(self):
+        # arrange
+        tags = ['bird', 'fineline']
+        vector_file = "tests/test2.jpg"
+        png_file = "tests/test2.jpg"
+        item_id = self.product_manager.add_product(tags, vector_file, png_file)
+
+        # act
+        self.product_manager.delete_product(item_id)
+
+        # assert
+        self.assertTrue(setup_test.assertCount(
+            self.database_connector, 'Items', 0))
+        self.assertTrue(setup_test.assertCount(
+            self.database_connector, 'Tags', 0))
+
+    def test_get_all_products(self):
+        # arrange
+        setup_test.test_items_tags_setup(self.item_manager, self.tag_manager)
+
+        # act
+        all_products = self.product_manager.get_all_products()
+
+        # assert
+        self.assertEqual(len(all_products), 4)
+
+    def test_get_all_matching_products(self):
+        # arrange
+        setup_test.test_items_tags_setup(
+            self.item_manager, self.tag_manager)
+
+        # act
+        all_matching_products = self.product_manager.get_all_matching_products(
+            "bir")
+
+        # assert
+        self.assertEqual(len(all_matching_products), 3)
+
+    def tearDown(self):
+        setup_test.tear_down_database_setup(self.database_connector)
+        bucket = self.s3_resource.Bucket(self.bucket)
+        for key in bucket.objects.all():
+            key.delete()
+        bucket.delete()
+
+    @classmethod
+    def tearDownClass(self):
+        self.database_connector = None
+        self.item_manager = None
+        self.tag_manager = None
+        self.image_manager = None
+        self.product_manager = None
```

Comment:
I'm not sure there is much use of settings these to None?

Reply:
deleted